aoservicelisting.civix.php

```diff
@@ -353,11 +347,24 @@ function setStatus($oldStatus = NULL, $cid, $submitValues) {
       $key = preg_grep('/^' . STATUS . '_[\d]*/', $submitKeys);
       $newStatus = reset($key);
       if ($oldStatus != "Approved" && CRM_Utils_Array::value($newStatus, $submitValues) == 'Approved') {
+        // Fetch primary contact.
+        $relationship = civicrm_api3('Relationship', 'get', [
+          'contact_id_b' => $cid,
+          'relationship_type_id' => PRIMARY_CONTACT_REL,
+          'return' => 'contact_id_a',
+        ]);
+        if ($relationship['count'] > 0 && !empty($relationship['values'][$relationship['id']]['contact_id_a'])) {
+          $primaryCid = $relationship['values'][$relationship['id']]['contact_id_a'];
+        }
         // Create drupal account if not exists.
-        self::createUserAccount($cid);
+        if (!self::createUserAccount($primaryCid)) {
+          // Set status message indicating that user account creation was unsuccessful.
+          $userUrl = CRM_Utils_System::url('civicrm/contact/view/useradd', "reset=1&action=add&cid=$primaryCid");
+          CRM_Core_Session::setStatus(ts('There was an error creating the user account. Please proceed to add one here: %1', [1 => $userUrl]), ts('Warning'), 'alert');
```

Comment:
@Edzelopez I tested this  by altering the primary contact for one of my test records to include an `=` in the last name. I can confirm that no user account was created, However no pop box showed indicating there was an error creating the user.